backend/app/api/endpoints/health.py

```
@@ -59,12 +93,137 @@ def readiness_check(db: Session = Depends(get_db)):
+
+    Returns:
+        Success status and message
+    """
+    # Validate status value
+    try:
+        new_status = ServiceStatus(request.status)
+    except ValueError:
+        raise HTTPException(
+            status_code=400,
+            detail=f"Invalid status value: {request.status}. Must be 'healthy' or 'draining'",
+        )
```

Ruff (0.14.8)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)